src/pages/keyword-generator/KeywordGenerator.vue

```diff
@@ -6,24 +6,55 @@
           <div class="ma-header">
             <span>Keyword Generator</span>
           </div>
-          <textarea v-model="inputValue" placeholder="Enter text here..." rows="6" class="input-textarea"></textarea>
+          <MaInput
+            v-model:value="inputValue"
+            type="textarea"            
+            size="large"
+            placeholder="Enter text here..."            
+            rows="6"
+          >
+            <template #title>Your Text</template>
+          </MaInput>          
+          <MaInput
+            v-model:value="excludedWords"
+            type="textarea"            
+            size="large"
+            placeholder="Enter words to exclude (space-separated)..."
+            hint-text="You may want to exclude 'is', 'a', 'an', 'the'."
+            rows="3"
+          >
+
+            <template #title>Exclude Words</template>
+          </MaInput>
+          <div class="multi-select-container">
+            <label>Select N-Grams to Generate:</label>
+            <div class="checkbox-group">
+              <div v-for="n in MAX_N_GRAMS_COUNT" :key="n">
+                <input type="checkbox" :id="'ngram-' + n" :value="n" v-model="selectedNGrams" />
+                <label :for="'ngram-' + n">{{ n }}-gram</label>
+              </div>
+            </div>
+          </div>
           <button @click="generateNGrams" class="custom-button">Generate N-Grams</button>
         </div>
         <div v-if="Object.keys(nGrams).some(key => nGrams[key].length)" class="keywords-output">
           <div class="ma-header">
-            <span>Generated Keywords</span>
+            <span>Generated Tags</span>
           </div>
           <div class="filter-container">
-            <label for="ngram-filter">Filter by:</label>
-            <select id="ngram-filter" v-model="selectedNGram" @change="filterKeywords" class="custom-select">
-              <option v-for="(value, key) in nGrams" :key="key" :value="key">{{ key }}</option>
-            </select>
+            <label>Filter by:</label>
+            <div class="checkbox-group">
+              <div v-for="(value, key) in sortedNGrams" :key="key">
+                <input type="checkbox" :id="'filter-' + key" :value="key" v-model="selectedNGramFilters" />
```

Comment:
We can use the checkbox component in the action-kit instead of the native `checkbox` element.